homeassistant/components/enocean/switch.py

```diff
@@ -49,7 +49,7 @@ def __init__(self, dev_id, dev_name, channel):
         self._on_state = False
         self._on_state2 = False
         self.channel = channel
-        self._attr_unique_id = f"{combine_hex(dev_id)}"
+        self._attr_unique_id = f"{combine_hex(dev_id)}{self.channel}"
```

Comment:
Please also migrate existing entity entries for the Enocean switch platform to the new unique_id pattern.

You can look up an existing entity entry with this method.

https://github.com/home-assistant/core/blob/8ccb008834638b847643b8a78b4c605b421d6bb2/homeassistant/helpers/entity_registry.py#L272-L275

Also add a test to check that the migration works.